src/main/kotlin/build/buf/gradle/BufExtension.kt

```diff
@@ -70,6 +76,22 @@ open class BufExtension {
     }
 }
 
+enum class AvailablePublicationFileExtension(val extension: String) {
+    BINPB("binpb"),
+    BINPB_GZ("binpb.gz"),
+    BINPB_ZST("binpb.zst"),
```

Comment:
Is it worth considering compression separately? That way if a new compression format is introduced or one is deprecated, then many fewer pieces have to move to reflect that deprecation.

Thinking about a potential API:
``` kotlin
buf {
    build {
        format = PublicationFormat.JSON
        compression = CompressionFormat.ZST
    }
}

Reply:
Change to define a format and compression method separately.